ws/ShardManager.ts

```diff
@@ -0,0 +1,35 @@
+import DiscordWebsocket, { ConnectionProperties } from './DiscordWebsocket';
+
+interface BotGateway {
+  session_start_limit: {
+    max_concurrency: number;
+    remaining: number;
+    reset_after: number;
+    total: number;
+  };
+  shards: number;
+  url: string;
+}
+
+export default class ShardManager {
+  private shards: Map<number, DiscordWebsocket> = new Map();
+
+  constructor(private token: string, private intents: number, private connProps: Partial<ConnectionProperties> = {}) {
+    this.initialize();
+  }
+
+  private async getBotGateway() {
+    const botGateway: BotGateway = await (await fetch('https://discord.com/api/v8/gateway/bot')).json();
```

Comment:
I dislike this ayntax with the double await.

Reply:
What do you suggest instead then?

Reply:
Await a `.then` pointing towards json

Reply:
or at least await on 2 separate lines. But I agree with Bsian solution, feels like it would be the cleanest in this situation